courses/intro-to-ada/intro_to_ada.rst

```diff
@@ -198,253 +207,427 @@ There are several note worthy things in the above program:
    create multiple lines of comments in Ada is by using :ada:`--` on each
    line. For example:
 
+.. admonition:: In other languages
+
+    Procedures ares similar to functions in C or C++ that return :c:`void`.
+    We'll see later how to declare functions in Ada.
+
 .. code-block:: ada
     :class: ada-nocheck
 
     --  We start a comment in this line...
     --  and we continue on the second line...
 
-Imperative language - If/Else
------------------------------
-
-Ada has an if statement. It is pretty unsurprising in form and function:
+Here is a minor variant of the "Hello, World" example:
 
 .. code-block:: ada
 
     with Ada.Text_IO; use Ada.Text_IO;
 
     procedure Greet is
-       I : Integer := 1;
     begin
-       loop
-          if I = 5 then
-             Put_Line ("Hello, World!");
-          end if;
-          I := I + 1;
-       end loop;
+       --  Print "Hello, World!" to the screen
+       Put_Line ("Hello, World!");
     end Greet;
 
-As for the while loop, the Boolean condition must be of strict type
-:ada:`Boolean`. Every relational operator in Ada returns a :ada:`Boolean`
-by default.
+This version utilizes an Ada feature known as a :ada:`use` clause, which has
+the the form ``use`` *package-name*. As illustrated by the call on
+:ada:`Put_Line`, the effect is that entities from the named package can be
+referenced directly, without the *package-name.* prefix.
+
+Imperative language - If/Then/Else
+----------------------------------
+
+This section describes Ada's ``if`` statement and introduces several other
+fundamental language facilities including integer I/O, data declarations,
+and subprogram parameter modes.
+
+Ada's ``if`` statementis pretty unsurprising in form and function:
 
 .. code-block:: ada
 
     with Ada.Text_IO; use Ada.Text_IO;
-    procedure Greet is
-       I : Integer := 0;
+    with Ada.Integer_Text_IO; use Ada.Integer_Text_IO;
+
+    procedure Check_Positive is
+       N : Integer;
     begin
-       loop
-          if I = 5 then
-             exit;
-             --  Exit can be unconditional
-          elsif I = 0 then
-             Put_Line ("Starting...");
-          else
-             Put_Line ("Hello, World!");
-          end if;
-          I := I + 1;
-       end loop;
-    end Greet;
+       Put ("Enter an integer value: ");  -- Put a String
+       Get (N);  -- Read in an integer value
+       if N>0 then
+          Put (N);  --Put an Integer
+          Put_Line (" is a positive number");
+       end if;
+    end Check_Positive;
+
+The ``if`` statement minimally consists of the reserved word :ada:`if`, a condition
+(which must be a boolean value), the reserved word :ada:`then` and a non-empty
+sequence of statements (the ``then`` part) which is executed if the condition
+evaluates to True, and a terminating :ada:`end if`.
+
+This example declares an Integer variable N, prompts the user for an Integer,
```

Comment:
"Integer", on the other hand, should not be capitalized if it's not directly referring to the type. If it is, however, then the ada-role is missing.